lib/generators/decidim/install_generator.rb

```diff
@@ -83,6 +75,52 @@ def test_mail_host
         end
       end
 
+      def smtp_environment
+        inject_into_file "config/environments/production.rb",
+                         after: "config.log_formatter = ::Logger::Formatter.new" do
+          %(
+
+  config.action_mailer.smtp_settings = {
+    :address        => Rails.application.secrets.smtp_address,
+    :port           => Rails.application.secrets.smtp_port,
+    :authentication => Rails.application.secrets.smtp_authentication,
+    :user_name      => Rails.application.secrets.smtp_username,
+    :password       => Rails.application.secrets.smtp_password,
+    :domain         => Rails.application.secrets.smtp_domain,
+    :enable_starttls_auto => Rails.application.secrets.smtp_starttls_auto,
+    :openssl_verify_mode => 'none'
+  }
+
+  if Rails.application.secrets.sendgrid
+    config.action_mailer.default_options = {
+      "X-SMTPAPI" => {
+        filters:  {
+          clicktrack: { settings: { enable: 0 } },
+          opentrack:  { settings: { enable: 0 } }
+        }
+      }.to_json
+    }
+  end
+          )
+        end
+      end
+
+      def smtp_secrets
+        inject_into_file "config/secrets.yml",
```

Comment:
We should probably provide a custom `secrets.yml` file in the future instead of just appending stuff.